jobserver/releases.py

```diff
@@ -103,3 +104,15 @@ def extract_upload(upload_dir, archive):
     actual_dir.parent.mkdir(exist_ok=True, parents=True)
     archive.extractall(actual_dir)
     return actual_dir
+
+
+def latest_files(releases):
+    """
+    Gets the latest version of each file across the given Releases
+    """
+
+    files = ReleaseFile.objects.filter(release__in=releases).order_by(
+        "-release__created_at"
+    )
+
+    return list(unique_everseen(files, key=lambda file: file.name))
```

Comment:
I am assuming we do this in python because of figuring out how to do it via ORM is tricky without DISTINCT ON support?

I think that's fine - we will never have many Releases, so all good.

However, I think adding an entire dependency just for single simple iterator is too much cost for not enough benefit.

I think it's simpler and more understandable just to do a simple for loop ourselves, and drop the dependency, e.g. something like this:

    seen = dict()
    for file in workspace.files.order_by("-release__created_at"):
        if file.name not in seen:
            seen[file.name] = file
    return list(seen.values())

Reply:
Yes, all the ORM/SQL solutions involved some hoop jumping and since we're unlikely to have that many Releases or ReleaseFiles in a Workspace (or even Project), doing it in Python should be fine.

I don't agree on the more-itertools argument, we've added many libraries for a single function call previously (first, pytest-subtests, incuna-mail, slack-sdk, etc) when we could have built many of those ourselves!

I'll swap it out for this PR though.

Reply:
So, I'm not advocating a general rule here, it's just in this specific case, the cost of maintaining an extra dependency is not worth it IMO, since the implementation is trivial.

If we are wanting to use other `more_iterools` functions in future, then sure no problem. But in the context of this change, it didn't seem worth it.

For slack and pytest-subtest, the implementation is not trivial, regardless of the fact its only one function. And we'd expect to use pytest-subtests multiple times going forward.